### ISSUE TITLE
[Bullet] RayCast.get_collider_shape() doesn't actually return Shape ID

### ISSUE THREAD

**Issue by manglemix** [godotengine/godot#37985]
Godot_v3.2-stable_win64

Windows 10 version 1809

The documentation states that RayCast.get_collider.shape() returns the Shape ID of the CollisionShape that the RayCast intersected. However, it doesn't seem to return the correct number.

var target = $RayCast.get_collider()
var shape = target.shape_owner_get_owner(target.shape_find_owner($RayCast.get_collider_shape()))

The above code in my project causes this error:
E 0:00:06.707   shape_find_owner: Index p_shape_index = 522 is out of bounds (total_subshapes = 1).
  <C++ Source>  scene/3d/collision_object.cpp:328 @ shape_find_owner()
  <Stack Trace> Bullet.gd:54 @ _process()

After asking for help in the Godot Discord Server, a Godot dev suggested that get_collider_shape() was returning the Shape RID instead. They suggested I check by using Godot Physics Engine, which after testing caused no errors

**Comment by akien-mga** [comment]
This should either be clarified in the documentation (`s/ID/RID/`), or the API could be changed so that it actually returns the shape index.

It might be worth making things consistent with KinematicCollision's API:
```
scene/3d/physics_body.cpp
1487:Object *KinematicCollision::get_collider_shape() const {
1500:int KinematicCollision::get_collider_shape_index() const {
```

While noting that in 2D there's also:
```
servers/physics_2d_server.cpp
495:int Physics2DTestMotionResult::get_collider_shape() const {
```
which I *think* returns a shape index, but I didn't check.

**Comment by pouleyKetchoupp** [comment]
Seems like it's the same as #39767, fixed in #42881

**Comment by pouleyKetchoupp** [comment]
@JakSparro98 Indeed, the issue actually still exists for raycasts. It was reported in #47989 with a minimal test project so I'll leave that one open.
